gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemIntegrationTest.java

```diff
@@ -1509,6 +1512,170 @@ public void testThreadTraceEnabledRename() throws Exception {
     assertThat(ghfs.exists(dest)).isTrue();
   }
 
+  @Test
+  public void testHnBucketNonRecursiveDeleteOperation() throws Exception {
+    String bucketName = this.gcsiHelper.getUniqueBucketName("hn");
+    GoogleHadoopFileSystem googleHadoopFileSystem = createHnEnabledBucket(bucketName);
+    String bucketPath = "gs://" + bucketName;
+    try {
+      googleHadoopFileSystem.mkdirs(new Path("/A/"));
+      assertThrows(
+          "Cannot delete a non-empty directory",
+          java.nio.file.DirectoryNotEmptyException.class,
+          () -> googleHadoopFileSystem.delete(new Path(bucketPath), false));
+
+      // verify only "A/" folder exists
+      assertThat(numberOfFolders(googleHadoopFileSystem, bucketPath + "/A")).isEqualTo(1);
+
+      // delete A/ non recursively
+      googleHadoopFileSystem.delete(new Path(bucketPath + "/A"), false);
+
+      // check that on listing we get no folders for folder "A/"
+      assertThat(numberOfFolders(googleHadoopFileSystem, bucketPath + "/A")).isEqualTo(0);
+    } finally {
+      googleHadoopFileSystem.delete(new Path(bucketPath));
+    }
+  }
+
+  @Test
+  public void testHnBucketRecursiveDeleteOperationOnBucket() throws Exception {
+    String bucketName = this.gcsiHelper.getUniqueBucketName("hn");
+    String bucketPath = "gs://" + bucketName;
+    GoogleHadoopFileSystem googleHadoopFileSystem = createHnEnabledBucket(bucketName);
+    createResources(googleHadoopFileSystem);
+    assertThat(numberOfFolders(googleHadoopFileSystem, "gs://" + bucketName)).isEqualTo(22);
+    assertThrows(
+        "Cannot delete a non-empty directory",
+        java.nio.file.DirectoryNotEmptyException.class,
+        () -> googleHadoopFileSystem.delete(new Path(bucketPath), false));
+
+    // delete bucket
+    googleHadoopFileSystem.delete(new Path(bucketPath), true);
+
+    // TODO: flaky as maybe deletion of bucket is asynchronous and still ongoing so should we add a
```

Comment:
discuss

Reply:
Are there existing tests for non-HN and how are we validating this?